### PR TITLE
Feature/ottobroker

### PR DESCRIPTION
**Pull request by lhillmer**

### CODE REVIEW

File: bot/broker.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+                            return 'Ottobot thanks you for your generousity, unregistered user'
+                        
+                        amount = Decimal(tip_info[1].split(':')[1])
+                        amount = self._exchange_rate * amount
+                        amount = Decimal(amount.quantize(Decimal('.01'), rounding=ROUND_HALF_UP))
+                        
+                        if amount != 0:
+                            self._db.broker_give_money_to_user(sender, amount, 'Tipping Ottobot')
+                            self._update_single_user(sender)
+                            return 'Ottobot winks at you, and walks away whistling. Your pockets feel heavier.'
```
Comment (nkhoit) [inline bot/broker.py]: Should return what your new balance is at.

File: bot/broker.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+                    receiver = tip_info[1].split(':')[0]
+                    if receiver == self._broker_id:
+                        try:
+                            self._get_user(sender)
+                        except Exception as e:
+                            return 'Ottobot thanks you for your generousity, unregistered user'
+                        
+                        amount = Decimal(tip_info[1].split(':')[1])
+                        amount = self._exchange_rate * amount
+                        amount = Decimal(amount.quantize(Decimal('.01'), rounding=ROUND_HALF_UP))
```
Comment (nkhoit) [inline bot/broker.py]: Shouldn't this round down?

Comment (lhillmer) [reply bot/broker.py]: yeah, I suppose that will prevent extra cents from being created? i'm not really sure how rounding for sub-cent values is *supposed* to work.

File: bot/broker.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    
+    async def _buy_regular_stock(self, user_id, user_display_name, symbol, per_stock_cost, quantity):
+        if not self.is_market_live():
+            raise Exception('Can\'t trade after hours')
+        # make the transaction, and report success
+        result = self._db.broker_buy_regular_stock(user_id, symbol, per_stock_cost, quantity)
+        if result is not None:
+            # if we succeeded, update the cached user
+            self._update_single_user(user_id)
+            return 'Congratulations {}, you\'re the proud new owner of {} additional {} stocks'.format(user_display_name, quantity, symbol)
```
Comment (nkhoit) [inline bot/broker.py]: This should note how much was spent.

File: bot/broker.py [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+        self._update_single_user(message_author.id)
+        new_user = self._user_cache[message_author.id]
+        return ('Welcome, {}. You have a starting balance of {}'.format(new_user.display_name, new_user.balance), True)
+
+    async def _handle_balance(self, command_args, message_author):
+        try:
+            user = self._get_user(message_author.id)
+
+            result_lines = [
+                ['Capital', user.balance, '']
```
Comment (nkhoit) [inline bot/broker.py]: I think this should just be "Cash" to denote the cash balance. But that's just my opinion, I don't think "Capital" is wrong per-say, just maybe ambiguous.

File: bot/broker.py (hunk 3 above)
Comment (nkhoit) [reply bot/broker.py]: Maybe in the form: Successfully bought {} @ {} for a total of {}.

**Comment by lhillmer** [comment]
This PR addresses #4 and #6

**Comment by lhillmer** [comment]
PR now also addresses #10, #11, and #13

**Pull request merged**